torch/testing/_internal/distributed/multi_threaded_pg.py

```diff
@@ -366,6 +399,7 @@ def worker(rank):
                         ),
                     )
                 )
+        ProcessLocalGroup.reset()
```

Comment:
Could you explain why we need the `reset` call? if every thread execute successfully, coll already set to None by `end_coll`, `terminate` won't get triggered?

Reply:
The reason is when thread fails `end_coll` may not be called (all threads throws exception after capturing the `terminate` event. So `coll` can be either None (successful test) or the last executed `Collective` object (failed test  on rank 0) at the call site of  `ProcessLocalGroup.reset()`.